### PR TITLE
Add unit test for `TensorNetwork` module

### PR DESCRIPTION
**Pull request by Todorbsc**

### CODE REVIEW

File: test/TensorNetwork_test.jl [hunk 1]
```diff
@@ -25,6 +25,85 @@
...
+        listtensors = [Tensor(rand(2,3,4), (:i, :j, :k)), Tensor(rand(4,3,2), (:l, :j, :m))]
+        tn = TensorNetwork(listtensors)
+
+        similartn = similar(tn)
+
+        @test length(tensors(tn)) == length(tensors(similartn))
+        @test issetequal(inds(tn), inds(similartn))
+        @test all(tensors(similartn)) do tns
+            all(isapprox.(tns, 0; atol=1e-10))
+        end
```
[the remark this answers is not in the document]
Comment (mofeing) [reply test/TensorNetwork_test.jl:38]: Also, it would be nice to check that the tensors have the same indices

File: test/TensorNetwork_test.jl [hunk 2]
```diff
@@ -319,6 +415,26 @@
...
 
+    @testset "Base.in" begin
+        @testset "by tensor" begin
+            tensor1 = Tensor(rand(3, 4), (:i, :j))
+            tensor2 = Tensor(rand(4, 5), (:j, :k))
+            tn = TensorNetwork([tensor1, tensor2])
+
+            # Broadcast not working
+            @test in(tensor1, tn)
+            @test in(tensor2, tn)
```
Comment (jofrevalles) [inline test/TensorNetwork_test.jl:426]: You can use `Base.Fix2` if you want to fix the second argument for broadcasting in the first:
```suggestion
            @test all(broadcast(Base.Fix2(in, tn), [tensor1, tensor2]))
```

File: test/TensorNetwork_test.jl [hunk 3]
```diff
@@ -319,6 +415,26 @@
...
+        end
+        @testset "by symbol" begin
+            indices = (:i, :j)
+            tn = TensorNetwork([Tensor(rand(3, 4), indices)])
+
+            # Broadcast not working
+            @test in(indices[1], tn)
+            @test in(indices[2], tn)
```
Comment (jofrevalles) [inline test/TensorNetwork_test.jl:434]:
```suggestion
            @test all(broadcast(Base.Fix2(in, tn), [tensor1, tensor2]))
```

**Review by jofrevalles — CHANGES_REQUESTED** [review]

File: test/TensorNetwork_test.jl (hunk 2 above)
Comment (mofeing) [reply test/TensorNetwork_test.jl:426]: I would write it like this:
```suggestion
            @test all(∈(tn), [tensor1, tensor2])
```

File: test/TensorNetwork_test.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (Todorbsc) [reply test/TensorNetwork_test.jl]: Done! Check the new test

File: test/TensorNetwork_test.jl [hunk 4]
```diff
@@ -25,6 +25,83 @@
...
 
+    @testset "Base.similar" begin
+        listtensors = [Tensor(rand(2, 3, 4), (:i, :j, :k)), Tensor(rand(4, 3, 2), (:l, :j, :m))]
+        tn = TensorNetwork(listtensors)
+
+        similartn = similar(tn)
+
+        @test length(tensors(tn)) == length(tensors(similartn))
+        @test issetequal(inds(tn), inds(similartn))
+        @test all(map((tns,simtns)-> inds(tns)==inds(simtns), tensors(tn), tensors(similartn)))
```
Comment (mofeing) [inline test/TensorNetwork_test.jl:36]: This is more readable:
```suggestion
        @test all(splat(==), zip(tensors(tn), tensors(similartn)))
```

**Review by mofeing — CHANGES_REQUESTED** [review]
Okay, just need to format the code with `JuliaFormatter` and it would be done.

File: src/TensorNetwork.jl [hunk 5]
```diff
@@ -622,7 +622,7 @@ end
 
 function LinearAlgebra.lu!(tn::TensorNetwork; left_inds=Symbol[], right_inds=Symbol[], kwargs...)
     tensor = tn[left_inds ∪ right_inds...]
-    L, U = lu(tensor; left_inds, right_inds, kwargs...)
-    replace!(tn, tensor => TensorNetwork([L, U]))
+    L, U, P = lu(tensor; left_inds, right_inds, kwargs...)
+    replace!(tn, tensor => TensorNetwork([P, L, U]))
```
Comment (mofeing) [inline src/TensorNetwork.jl:626]: @Todorbsc why are you returning here `P`? We deliberately discard `P`.

Comment (mofeing) [reply src/TensorNetwork.jl:626]: Is it completely needed to reconstruct `tensor`?

**Review by mofeing — APPROVED** [review]

**Comment by jofrevalles** [comment]
Do you want to merge this, @mofeing ?

**Comment by mofeing** [comment]
Not yet. I think @Todorbsc wants to finish some tests more.

**Review by mofeing — APPROVED** [review]

**Comment by Todorbsc** [comment]
As long as you have no more suggestions/corrections, it is ready for merge.

**Pull request merged**